language/tools/test-generation/src/lib.rs

```diff
@@ -104,9 +102,17 @@ fn execute_function_in_module(
         let mut context =
             TransactionExecutionContext::new(*MAXIMUM_NUMBER_OF_GAS_UNITS, &data_cache);
         let gas_schedule = runtime.load_gas_schedule(&mut context, &data_cache)?;
-        let mut txn_executor =
-            TransactionExecutor::new(&gas_schedule, &data_cache, TransactionMetadata::default());
-        txn_executor.execute_function(&runtime, &module_id, &entry_name, args)
+        let txn_data = TransactionMetadata::default();
```

Comment:
@tzakian take a look at this though I think it should be straightforward

Reply:
LGTM!